### ISSUE TITLE
Radiobuttons are duplicated during `reset_choices` in Qt <5.15.2

### ISSUE THREAD

**Issue by maweigert** [napari/magicgui#197]
**Describe the bug**

When creating a napari plugin and adding `RadioButtons` widget with `magicgui`, every radiobutton appears duplicated (see screenshot).  

**To Reproduce**

Create a simple plugin consisting out of a single group of RadioButtons

```python 

from napari_plugin_engine import napari_hook_implementation
from magicgui import magicgui

def wrapper():
    @magicgui (
        choices = dict(widget_type='RadioButtons', label='Choices',
                   orientation='horizontal', choices=["A","B","C"])
    )
    def widget (
            choices
    ):
        pass
    return widget


@napari_hook_implementation
def napari_experimental_provide_dock_widget():
    return wrapper, {'name': 'BUG'}


if __name__ == '__main__':

    import napari

    with napari.gui_qt():
        viewer =  napari.Viewer()

        key = ("BUG", "BUG") 

        viewer.window._add_plugin_dock_widget(key)

```

**Screenshots**

<img width="1007" alt="Screenshot 2021-03-24 at 16 18 01" src="https://user-images.githubusercontent.com/11042162/112335846-de860f00-8cbc-11eb-9102-958172b20577.png">

**Environment (please complete the following information):**

```
OS        OSX 10.15.7
QT        5.15.1
napari    0.4.6
magicgui  0.2.8.dev14+g9590805
```

**Comment by tlambert03** [comment]
hmmm... I can't reproduce this (tried both with napari 0.4.6 and the newest version of napari)

can you try this slightly more minimal example:

```python
from magicgui import magicgui
import napari

@magicgui(
    choices=dict(
        widget_type="RadioButtons",
        label='Choices',
        orientation="horizontal",
        choices=["A", "B", "C"],
    )
)
def widget(choices):
    ...

viewer = napari.Viewer()
viewer.window.add_dock_widget(widget, area="right")
napari.run()
```
<img width="1269" alt="Untitled" src="https://user-images.githubusercontent.com/1609449/112337967-c9958580-8c94-11eb-8101-24d675ef13a4.png">


if you still have the error, can you give me the full output of `pip list`?

side note: in napari master and in the upcoming release, the private `_add_plugin_dock_widget` has been made [public `add_plugin_dock_widget`](https://github.com/napari/napari/blob/5c151e5553fa98a41d7140bd05056acdcb6dc86d/napari/_qt/qt_main_window.py#L735-L748), and it accepts the name of the plugin

**Comment by tlambert03** [comment]
aha, ok, looks like this is a Qt 5.15.1 issue...  i downgraded and saw it (both pyside and pyqt5).  will take a look

**Comment by maweigert** [comment]
> aha, ok, looks like this is a Qt 5.15.1 issue... i downgraded and saw it (both pyside and pyqt5). will take a look. (but, in the meantime, if you use ≥5.15.2 you should be ok)

Indeed, that seemed to be the problem. Upgrading to `5.15.4` solved the issue. Thanks a lot!

**Comment by tlambert03** [comment]
I think I will keep this open though, since I'm also seeing this on pyqt5-5.14.2 ... and those are definitely all "supported" versions.  thanks for reporting!

**Comment by tlambert03** [comment]
upon a little further investigation... I think we need to move this to napari.  I don't see this if I run magicgui independently of napari: 

```python
from magicgui import magicgui


@magicgui(
    choices=dict(
        widget_type="RadioButtons",
        orientation="horizontal",
        choices=["A", "B", "C"],
    )
)
def widget(choices):
    ...


widget.show(run=True)
```

so there must be a bug in the reparenting of widgets when napari adds a dock widget

**Comment by maweigert** [comment]
Sure, thanks for the super fast response!

**Comment by tlambert03** [comment]
it's a magicgui bug after all!  turns out this is a bug in `widget.reset_choices`: categorical widgets like this can declare their options as a callable (in addition to a list), and when a widget gets reparented (like when adding to the napari window) the choices are refreshed.  The radiobuttons qt implementation has a bug in reset_choices, such that buttons are duplicated.  MRE:

```
from magicgui import magicgui

@magicgui(choices=dict(widget_type="RadioButtons", choices=["A", "B", "C"]))
def widget(choices):
    pass

widget.reset_choices()
widget.show(run=True)
```

**Comment by tlambert03** [comment]
fixed in #198 ... will release another patch release (0.2.9) shortly
